downloads/swan-lake-release-notes/swan-lake-beta3/RELEASE_NOTE.md

```diff
@@ -0,0 +1,772 @@
+
+The following will now result in errors.
+
+```ballerina
+const int INTEGER = INTEGER; // Compilation error.
+
+public function main() {
+    string s = let string[] arr = [s] in arr[0]; // Compilation error.
+}
+```
+
+- Type narrowing is now reset on any assignment to a narrowed variable. Previously, type narrowing was not reset if the static type of the new value was a subtype of the narrowed type. This was a deviation from the specification.
+
+```ballerina
+public function main() {
+    int|string v = 0;
+
+    if v is int {
+        int x = v; // Valid.
+
+        // Now, the type is reset to `int|string` even though `1` belongs to `int`.
+        v += 1;
+
+        int y = v; // Invalid, will now result in an error.
+    }
+}
+```
+
+To view all bug fixes, see the [GitHub milestone for Swan Lake Beta3](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22Ballerina+Swan+Lake+-+Beta3%22+label%3AType%2FBug+label%3ATeam%2FCompilerFE).
+
+### Runtime Updates
+
+#### Improvements
+
+##### Improved Configurable Variables to Support XML Types Through TOML Syntax
+
+The `configurable` feature is improved to support variables with XML types through the TOML syntax.
+
+For example, if the XML-typed configurable variables are defined in the following way,
+
+``` ballerina
+configurable xml xmlVar = ?;
+```
+the values can be provided in the `Config.toml` file as follows.
+
+
+```toml
+xmlVar = "<book><name>Sherlock Holmes</name></book>"
+```
+
+##### Improved Configurable Variables to Support Additional Fields and Rest Fields for Records
+
+The `configurable` feature is improved to support additional fields and rest fields in record variables through the TOML syntax.
+
+For example, if a configurable variable with open record type is defined in the following way,
+
+```ballerina
+type Person record {
+};
+
+configurable Person person = ?;
+```
+
+the values can be provided in the `Config.toml` file as follows.
+
+
+```toml
+[person]
+intVal = 22
+floatVal = 22.33
+stringVal = "abc"
+arrVal = [1,2,3]
+mapVal.a = "a"
+mapVal.b = 123
+```
+
+The additional fields that are created from the TOML values will have the following types.
+
+* TOML Integer - `int`
+* TOML Float - `float`
+* TOML String - `string`
+* TOML Boolean - `boolean`
+* TOML Table - `map<anydata>`
+* TOML Table array - `map<anydata>[]`
+
+Similarly, if a configurable variable with a record type that contains a rest field is defined in the following way,
+
+```ballerina
+public type Numbers record {|
+   int ...;
+|};
+
+configurable Numbers numbers = ?;
+```
+
+the values can be provided in the `Config.toml` file as follows.
+
+
+```toml
+num1 = 11
+num2 = 26
+```
+
+##### Improved the Printed Error Stacktrace to Include the Cause
+
+The error stack trace has been improved to include the error cause locations. Stack frames of the wrapped error causes are also added to the stack trace.
+
+E.g.,
+For the following example,
```

Comment:
Isn't it repetitive to have both `E.g.` and `following example`?

Reply:
Removed the `E.g.` part.